docs/dc-apps-performance-toolkit-user-guide-bitbucket.md

```diff
@@ -23,7 +32,211 @@ For simple spikes or tests, you can skip steps 1-2 and target any Bitbucket test
+{{% /warning %}}
+
+---
+
+### <a id="devappaction"></a>4. Develop and test app-specific action locally
+Data Center App Performance Toolkit has its own set of default test actions for Bitbucket Data Center: JMeter and Selenium for load and UI tests respectively.     
+
+**App-specific action** - action (performance test) you have to develop to cover main use cases of your application. Performance test should focus on the common usage of your application and not to cover all possible functionality of your app. For example, application setup screen or other one-time use cases are out of scope of performance testing.
+
+1. Define main use case of your app. Usually it is one or two main app use cases.
+1. Your app adds new UI elements in Bitbucket Data Center - Selenium app-specific action has to be developed.
+1. Your app introduces new endpoint or extensively calls existing Bitbucket Data Center API - JMeter app-specific actions has to be developed.  
+
+
+{{% note %}}
+We strongly recommend developing your app-specific actions on the development environment to reduce AWS infrastructure costs.
+{{% /note %}}
+
+#### Example of app-specific Selenium action development
+You develop an app that adds some additional fields to specific types of Bitbucket issues. In this case, you should develop Selenium app-specific action:
+
+1. Extend example of app-specific action in `dc-app-performance-toolkit/app/extension/bitbucket/extension_ui.py`.  
+[Code example.](https://github.com/atlassian/dc-app-performance-toolkit/blob/master/app/extension/bitbucket/extension_ui.py)
+So, our test has to open app-specific issues and measure time to load of this app-specific issues.
+1. If you need to run `app_speicifc_action` as specific user uncomment `app_specific_user_login` function in [code example](https://github.com/atlassian/dc-app-performance-toolkit/blob/master/app/extension/bitbucket/extension_ui.py). Note, that in this case `test_1_selenium_custom_action` should follow just before `test_2_selenium_z_log_out` action.
+1. In `dc-app-performance-toolkit/app/selenium_ui/bitbucket-ui.py`, review and uncomment the following block of code to make newly created app-specific actions executed:
+``` python
+# def test_1_selenium_custom_action(webdriver, datasets, screen_shots):
+#     app_specific_action(webdriver, datasets)
+```
+
+4. Run toolkit with `bzt bitbucket.yml` command to ensure that all Selenium actions including `app_specific_action` are successful.
+
+## <a id="mainenvironmententerprise"></a> Enterprise-scale environment
+
+After adding your custom app-specific actions, you should now be ready to run the required tests for the Marketplace Data Center Apps Approval process. To do this, you'll need an **enterprise-scale environment**.
+
+## <a id="instancesetup"></a>1. Set up an enterprise-scale environment Bitbucket Data Center on AWS
```

Comment:
This should be ### hashtags and number 5. Otherwise, the table of content and numeration are borked.
 Also, the same applicable for next sections.